baya/backend.py

```diff
@@ -33,3 +48,31 @@ def get_all_permissions(self, user, obj=None):
                     permissions.add("%s.%s" % (app, perm_name('delete')))
         user._baya_cached_all_permissions = permissions
         return permissions
+
+
+class ReconnectingLDAP(object):
+    """
+    An object that makes the standard ldap.initialize function use
+    ldap.ldapobject.ReconnectLDAPObject instead.
+
+    The django_auth_ldap library does not provide a simple way to
+    override how the LDAP connection is created. The actual method
+    that initiates the connection is on the _get_connection method
+    on the hidden _LDAPUser class.
+
+    Instances of this class take the original ldap module instance
+    and provide the same set of attributes, except that the initialize
+    attribute returns ReconnectLDAPObject instead of the original
+    initialize function (which just instantiates a SimpleLDAPObject).
+    """
+    retry_max = 6
+
+    def __init__(self, original_module):
+        self._original_module = original_module
+
+    def __getattr__(self, name):
+        return getattr(self._original_module, name)
```

Comment:
I know it's standard python, but I can never remember what the rarely-used subtleties of `__getattribute__`, `__getattr__` and `getattr` are. (I had to look it up.) Maybe add a comment reminding the reader that `__getattr__` is called when an attribute is not found in the instance `__dict__` or in the class MRO.

Reply:
Good call, I had to look it up myself.